repos/re-theme/src/styleInjector/injectHelpers.js

```diff
@@ -0,0 +1,222 @@
+
+/**
+ * Converts a camelCase style rule into a hyphenated style rule
+ * <br/>Caches the response to make future conversions faster
+ * @function
+ * @param {string} str - camelCase style rule rule
+ *
+ * @returns {string} - Hyphenated style rule
+ */
+export const hyphenator = rule => {
+  if (hyphenCache.hasOwnProperty(rule)) return hyphenCache[rule]
+
+  const hRule = rule.replace(uppercasePattern, toHyphenLower)
+  return (hyphenCache[rule] = msPattern.test(hRule) ? '-' + hRule : hRule)
+}
+
+/**
+ * Creates a hash from a passed in string consistently
+ * <br/>Not intended to be secure
+ * <br/>Value comes from being a pure function
+ * <br/>Given the same input, it will always return the same output
+ * <br/>There is no expectation to convert back from the hash to the original string
+ * @function
+ * @param {string} str - String to be hashed
+ * @param {number=} maxLength - Max length of the returned hash
+ *
+ * @returns {string} - Hashed version of the string
+ */
+export const hashString = (str, maxLength=0) => {
+  if (!isStr(str) || str.length == 0) return 0
+
+  str = str.split('').reverse().join('')
+
+  let hash = 0
+  for (let i = 0; i < str.length; i++) {
+    const char = str.charCodeAt(i)
+    hash = ((hash<<5) - hash) + char
+    // Convert to positive 32bit integer
+    hash = `${ Math.abs(hash & hash) }`
+  }
+
+  return maxLength ? hash.slice(0, maxLength) : hash
+}
+
+/**
+ * Default style rules to not convert to CSS
+ * @Object
+ */
+const defFilterRules = [
+  'aspectRatio',
+  'elevation',
+  'overlayColor',
+  'resizeMode',
+  'tintColor',
+  'backgroundSize',
+  'animationKeyframes',
+  'placeholderTextColor',
+  'pointerEvents',
+  'scrollbarWidth',
+]
+
+/**
+ * Filters out styles that require extra help to convert to CSS
+ * <br/>In these cases, we let react-native-web handel through the style attribute
+ * @function
+ * @param {Object} style - Object containing style rules
+ * @param {Array} filter - Custom style rules to filter out
+ *
+ * @returns {Object} - Contains separated objects with filtered styles, non-filtered styles
+ */
+export const filterRules = (style, filter) => {
+  const toFilter = isArr(filter) ? defFilterRules.concat(filter) : defFilterRules
+  return {
+    style: omitKeys(style, toFilter),
+    filtered: pickKeys(style, toFilter)
+  }
+}
+
+/**
+ * Creates a unique selector based on the passed in className and cssString
+ * @function
+ * @param {string} className - Original className used as a css selector
+ * @param {string} cssString - Css rules for the className in string format
+ *
+ * @returns {string} - Hashed version of the string
+ */
+export const getSelector = (className, cssString) => {
+  const selector = !exists(className)
+    ? false
+    : isArr(className)
+      ? className.filter(cls => cls).join('.').trim()
+      : isStr(className) && className.split(' ').join('.').trim()
+
+  return selector
+    ? `.${selector.trim()}.keg-${hashString(cssString)}`.trim()
```

Comment:
I think there are maybe some redundant trim calls, since you trim `selector` here and also on lines `150` and `151`.